Alignment/OfflineValidation/bin/GCPtrends.cc

```diff
@@ -0,0 +1,272 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include "TROOT.h"
+#include "TFile.h"
+#include "TChain.h"
+#include "TSelector.h"
+#include "TTreeReader.h"
+#include "TTreeReaderValue.h"
+#include "TTreeReaderArray.h"
+#include "TGraphErrors.h"
+#include "TH1D.h"
+#include "TObject.h"
+#include "TMath.h"
+#include "TString.h"
+
+#include "Alignment/OfflineValidation/interface/Trend.h"
+
+int main(int argc, char *argv[]) {
+  if (argc < 2) {
+    std::cout << "Usage of the program : GCPtrends [gcp_trend_config.json]" << std::endl;
+    std::cout << "gcp_trend_config.json : file with configuration in json format" << std::endl;
+    exit(1);
+  }
+
+  // parsing input file
+  std::string lumi_file(argv[1]);
+  boost::property_tree::ptree config_root;
+  boost::property_tree::read_json(lumi_file, config_root);
+
+  // configuration
+  TString outputdir = config_root.get<std::string>("outputdir", ".");
+  outputdir = TString(outputdir) + TString(outputdir.EndsWith("/") ? "" : "/");
+
+  bool usebadmodules = config_root.get<bool>("usebadmodules", true);
+
+  bool splitpixel = config_root.get<bool>("splitpixel", false);
+
+  Int_t trackerphase = config_root.get<int>("trackerphase");
+
+  TString lumitype = config_root.get<std::string>("trends.lumitype");
+
+  Int_t firstrun = config_root.get<int>("trends.firstrun");
+
+  Int_t lastrun = config_root.get<int>("trends.lastrun");
+
+  std::string lumibyrunfile = config_root.get<std::string>("lumibyrunfile", "");
+  std::ifstream lumifile(lumibyrunfile);
+  assert(lumifile.good());
+  const Run2Lumi lumi_per_run(lumibyrunfile, firstrun, lastrun, 1000);
+
+  std::map<TString, Int_t> comparison_map;
```

Comment:
I think C++ native types are generally preferred to ROOT types, but perhaps it's not mandatory here, as this is essentially a plotting macro.